### ISSUE TITLE
Implement date, currency, and number formatters(?)

### ISSUE THREAD

**Issue by mwistrand** [dojo/i18n#27]
Application developers, in addition to ICU message formatting, need a way to format dates, numbers, and currencies according to a specific locale. In my mind, it makes sense to create a separate module for date-related formatting, numerical formatting, and unit formatting: `date.ts` (dates and times), `number.ts` (numbers and currencies), and `unit.ts`.

However, given that the locale-specific CLDR data required for these operations are loaded within `i18n.ts`, any methods added to `dojo-i18n` would likely do nothing more than delegate to their Globalize.js counterparts. In this case, would we even need to supply them? Should we instead recommend using Globalize.js directly for these formatters?

**Comment by mwistrand** [comment]
The one benefit I can think of from providing our own methods is that it would remove the need to manually create new `Globalize` instances for locales. For example,

```typescript
import i18n from './i18n';

export function getDateFormatter(options: DateOptions, locale?: string): DateFormatter {
    const globalize = locale && locale !== i18n.locale ? new Globalize(locale) : Globalize;
    return globalize.dateFormatter(options);
}
```

**Comment by mwistrand** [comment]
Resolved by #28
